lib/validation/reply.spec.ts

```diff
@@ -0,0 +1,36 @@
+import { assertThrowsChain } from "../helpers.spec";
+import { validateMessageID } from "./reply";
+import { ValidationError } from "./validation-error";
+
+describe("./validation/reply", function () {
+  describe("#validateMessageID()", function () {
+    it("rejects undefined", function () {
+      assertThrowsChain(
+        () => validateMessageID(undefined),
+        ValidationError,
+        "Message ID undefined is invalid/malformed"
+      );
+    });
+
+    it("rejects null", function () {
+      assertThrowsChain(
+        () => validateMessageID(null),
+        ValidationError,
+        "Message ID null is invalid/malformed"
+      );
+    });
+
+    it("rejects empty strings", function () {
+      assertThrowsChain(
+        () => validateMessageID(""),
+        ValidationError,
+        "Message ID empty string is invalid/malformed"
+      );
+    });
+
+    it("allows dashes", function () {
+      validateMessageID("885196de-cb67-427a-baa8-82f9b0fcd05f");
+      validateMessageID("123e4567-e89b-12d3-a456-426614174000");
```

Comment:
It's worth noting that message IDs follow the following format: `xxxxxxxx-xxxx-4xxx-xxxx-xxxxxxxxxxxx`, so the second ID here is technically invalid.